File: tests/xcalablemp/not-implemented-yet/global-view/array/C/array6.c
/*testp014.c*/
/*array指示文のテスト：多次元分散+部分配列+ストライドあり*/
#include<xmp.h>
#include<mpi.h>
#include<stdio.h>    
static const int N=1000;
int a[N][N];
double b[N][N];
float c[N][N];
char *result;
#pragma xmp nodes p(4,*)
#pragma xmp template t(N,N)
#pragma xmp distribute t(block,block) onto p
#pragma xmp align a[i][j] with t(i,j)
#pragma xmp align b[i][j] with t(i,j)
#pragma xmp align c[i][j] with t(i,j)
int i,j;
int main(void){
#pragma xmp loop on t(i)
   for(i=0;i<N;i++){
      for(j=0;j<N;j++){
         a[i][j]=j*N+i;
         b[i][j]=(double)(j*N+i);
         c[i][j]=(float)(j*N+i);
      }
   }

#pragma xmp array on t(:)
   a[222:556:2][333:112:3] = 0;
#pragma xmp array on t(:)
   b[111:667:3][444:112:4] = 0.0;
#pragma xmp array on t(:)
   c[222:667:4][555:112:5] = 0.0;

   result = "OK";

#pragma xmp loop (j,i) on t(j,i)
   for(i=333;i<445;i=i+3){
      for(j=222;j<778;j=j+2){
         if(a[i][j] != 0){
            result="NG";
         }else{
            a[i][j] = j*N*i;
         }
      }
   }

#pragma xmp loop (j,i) on t(j,i)
   for(i=444;i<556;i=i+4){
      for(j=111;j<778;j=j+3){
         if(b[i][j] != 0){
            result = "NG";
         }else{
            b[i][j] = (double)j*N*i;
         }
      }
   }
#pragma xmp loop (j,i) on t(j,i)
   for(i=555;i<667;i=i+5){
      for(j=222;j<889;j=j+4){
         if(c[i][j] != 0){
            result = "NG";
         }else{
            c[i][j] = (float)j*N*i;
         }
      }
   }

#pragma xmp loop (j,i) on t(j,i)
   for(j=0;j<N;j++){
      for(i=0;i<N;i++){
         if(a[i][j] != j*N+i) result="NG";
         if(b[i][j] != (double)(j*N+i)) result="NG";
         if(c[i][j] != (float)(j*N+i)) result="NG";      
      }
   }
   printf("%d %s %s\n",xmp_node_num(),"testp014.c",result); 
   return 0;
}
      
         
      
   
         

File: tests/xcalablemp/NOT-IMPLEMENTED/23/array6.f90
! testp014.f
! array指示文のテスト：多次元分散 + 部分配列 + ストライドあり

      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
      integer a(N,N)
      real*4 b(N,N)
      real*8 c(N,N)
      character(len=2) result
!$xmp nodes p(4,*)
!$xmp template t(N,N)
!$xmp distribute t(block,block) onto p
!$xmp align a(i,j) with t(i,j)
!$xmp align b(i,j) with t(i,j)
!$xmp align c(i,j) with t(i,j)
      common /xxx/ b

!$xmp loop on t(i,j)
      do j=1, N
         do i=1, N
            a(i,j) = (j-1)*N+i
            b(i,j) = real((j-1)*N+i)
            c(i,j) = dble((j-1)*N+i)
         enddo
      enddo

!$xmp array on t(:,:)
      a(222:777:2,333:444:3) = 0
!$xmp array on t(:,:)
      b(111:777:3,444:555:4) = 0.0
!$xmp array on t(:,:)
      c(222:888:4,555:666:5) = 0.0

      result = 'OK'

!$xmp loop (i,j) on t(i,j)
      do j=333,444,3
         do i=222,777,2
            if(a(i,j) .ne. 0) then
               result = 'NG'
            else
               a(i,j) = (j-1)*N+i
            endif
         enddo
      enddo

!$xmp loop (i,j) on t(i,j)
      do j=444,555,4
         do i=111,777,3
            if(b(i,j) .ne. 0.0) then
               result = 'NG'
            else
               b(i,j) = real((j-1)*N+i)
            endif
         enddo
      enddo

!$xmp loop (i,j) on t(i,j)
      do j=555,666,5
         do i=222,888,4
            if(c(i,j) .ne. 0.0) then
               result = 'NG'
            else
               c(i,j) = dble((j-1)*N+i)
            endif
         enddo
      enddo

!$xmp loop (i,j) on t(i,j)
      do j=1, N
         do i=1, N
            if(a(i,j) .ne. (j-1)*N+i) result = 'NG'
            if(b(i,j) .ne. real((j-1)*N+i)) result = 'NG'
            if(c(i,j) .ne. dble((j-1)*N+i)) result = 'NG'
         enddo
      enddo

      print *, xmp_node_num(), 'testp014.f ', result

      end
